### PR TITLE
New rebuild_package script that can take folders as well as zip files

### PR DESCRIPTION
**Pull request by hossenlopp**
This is replacing the existing `add_json_to_package` task that is now broken due to changes elsewhere. This task focuses on just replacing the ELM XML and JSON based on the CQL in the package. It can take a folder or zip file and outputs a zip package adjacent to the existing one.

Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)

**Reviewer 1:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] You have tried to break the code

### CODE REVIEW

File: lib/tasks/rebuild_package.rake [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+      input_package_or_dir_path = Pathname.new(args[:input_package_or_dir_path])
+
+      if input_package_or_dir_path.directory?
+        puts "Rebuilding ELM of measure folder at #{args[:input_package_or_dir_path]}"
+        measure_path = input_package_or_dir_path
+      else
+        puts "Rebuilding ELM of measure package at #{args[:input_package_or_dir_path]}"
+        measure_path = Pathname.new(File.join('tmp', 'package_temp'))
+        FileUtils.rm_rf(measure_path) if measure_path.exist?
+        measure_path.mkdir
```
Comment (lukoktonos) [inline lib/tasks/rebuild_package.rake]: This should use `FileUtils.mkdir_p`

File: lib/tasks/rebuild_package.rake [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        Zip::ZipFile.open(input_package_or_dir_path) do |zip_file|
+          human_readable_path = zip_file.glob(File.join('**','**.html')).reject { |x| x.name.starts_with?('__MACOSX') }.first
+          zip_file.extract(human_readable_path, File.join(measure_path, File.basename(human_readable_path.name)))
+
+          # look at the human_readable to see if it has a directory it resides in
+          base_zip_directory = File.dirname(human_readable_path.name) if File.dirname(human_readable_path.name) != "."
+
+          # extract all other needed files
+          zip_file.glob(File.join('**','**.cql')).reject { |x| x.name.starts_with?('__MACOSX') }.each do |cql_file_path|
+            zip_file.extract(cql_file_path, File.join(measure_path, File.basename(cql_file_path.name)))
```
Comment (lukoktonos) [inline lib/tasks/rebuild_package.rake:37]: This line is erroring out when testing with CMSAWA_v5_6_Artifacts.zip . I think a comment noting that composite measures are not supported would be good.
```
Zip::DestinationFileExistsError: Destination 'tmp/package_temp/Hospice-2.0.000.cql' already exists
```

File: lib/tasks/rebuild_package.rake [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+          zip_file.glob(File.join('**','**.xml')).reject { |x| x.name.starts_with?('__MACOSX') }.each do |xml_file_path|
+            zip_file.extract(xml_file_path, File.join(measure_path, File.basename(xml_file_path.name)))
+          end
+        end
+
+      end
+
+      # translate_cql_to_elm
+      cql_files = Dir.glob(File.join(measure_path, '**.cql'))
+      elm_jsons, elm_xmls = CqlElm::CqlToElmHelper.translate_cql_to_elm(cql_files.map { |cql_path| File.read(cql_path) })
```
Comment (lukoktonos) [inline lib/tasks/rebuild_package.rake:48]: Does this require the cql-to-elm JAR to be running?

```
Errno::ECONNREFUSED: Failed to open TCP connection to localhost:8080 (Connection refused - connect(2) for "localhost" port 8080)
```

**Review by lukoktonos — CHANGES_REQUESTED** [review]
I have been unable to get this rake task to work--it may be my local environment isn't set up correctly.  See comments for the error's I'm seeing.

File: lib/tasks/rebuild_package.rake (hunk 3 above)
Comment (hossenlopp) [reply lib/tasks/rebuild_package.rake:48]: Yes. It requires the cql-to-elm to be running. I'll add a comment.

File: lib/tasks/rebuild_package.rake (hunk 2 above)
Comment (hossenlopp) [reply lib/tasks/rebuild_package.rake:37]: I did not want to touch the composite measure packages in this iteration. I'll add a comment.

File: lib/tasks/rebuild_package.rake (hunk 1 above)
Comment (hossenlopp) [reply lib/tasks/rebuild_package.rake]: Good call, this assumes the `tmp` directory exists. I know you like to wipe it out when changing branches.

**Review by lukoktonos — APPROVED** [review]

**Pull request merged**